auth/ante.go

```diff
@@ -55,12 +56,21 @@ type FeeCollector interface {
 	) sdk.Error
 }
 
+//
+// MainTxMsg tx hash
+//
+type MainTxMsg interface {
```

Comment:
Please use a better name for this struct.

Reply:
Suggestion: `BaseMsg`. Maybe we should make all our `msg` satisfy `BaseMsg` which would satisfy `sdk.Msg` as well.

Reply:
`BaseMsg` shows the wrong terminology here. This tx should satisfy the Main chain tx.